### PR TITLE
Add --record-to option to flutter tools

### PR DESCRIPTION
**Pull request by tvolkert**
This option will cause flutter tools to record all process
invocations that occur and serialize their stdout and stderr
to files that get added to a "recording" ZIP file. This is
part of an effort to be able to test flutter tools in a hermetic
environment.

As a side-benefit, this recording should prove an excellent
attachment to any bug report.

### CODE REVIEW

**Comment by tvolkert** [comment]
@danrubel @johnmccutchan @cbracken

File: packages/flutter_tools/lib/src/runner/flutter_command_runner.dart [hunk 1]
```diff
@@ -93,6 +94,14 @@ class FlutterCommandRunner extends CommandRunner<Null> {
             'Name of a build output within the engine out directory, if you are building Flutter locally.\n'
             'Use this to select a specific version of the engine if you have built multiple engine targets.\n'
             'This path is relative to --local-engine-src-path/out.');
+    argParser.addOption('record-to',
+        help:
+            'Enables recording of process invocations (including stdout and '
+            'stderr of all such invocations), and serializes that recording to '
+            'the specified location.\n'
+            'If the location is a folder, a ZIP file named `recording.zip` will'
```
Comment (cbracken) [inline packages/flutter_tools/lib/src/runner/flutter_command_runner.dart]: `s/folder/directory/` for consistency with other help strings.

**Review by cbracken — APPROVED** [review]
4992068468_df00990aa3

**Comment by cbracken** [comment]
@cbracken fails at `Load Diff`. Taking a look at the other 400 loc now :)

File: packages/flutter_tools/lib/src/runner/flutter_command_runner.dart (hunk 1 above)
Comment (tvolkert) [inline packages/flutter_tools/lib/src/runner/flutter_command_runner.dart]: Done

**Review by danrubel — APPROVED** [review]

**Comment by johnmccutchan** [comment]
LGTM. Very nice.

**Pull request merged**